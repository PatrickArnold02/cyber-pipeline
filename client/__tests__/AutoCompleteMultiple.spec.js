import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
// Mock PrimeVue components
import FloatLabel from 'primevue/floatlabel'
import IconField from 'primevue/iconfield'
import InputIcon from 'primevue/inputicon'
import AutoComplete from 'primevue/autocomplete'

// Import the component to test
import AutocompleteMultiple from '@/components/forms/AutocompleteMultiple.vue'

describe('AutoCompleteMultiple', () => {
  const props = {
    field: 'testField',
    label: 'Test Label',
    icon: 'pi pi-search',
    values: [
      { name: 'Apple' },
      { name: 'Banana' },
      { name: 'Orange' }
    ],
    valueLabel: 'name',
    disabled: false,
    modelValue: [],
    errors: {
      testField: [{ message: 'Field is required' }]
    }
  }

  it('renders correctly with provided props', () => {
    const wrapper = mount(AutocompleteMultiple, {
      props
    })

    expect(wrapper.find('label').text()).toBe(props.label)
    expect(wrapper.find('.pi-search').exists()).toBe(true)
  })

  it('filters autocomplete values based on search query', async () => {
    const wrapper = mount(AutocompleteMultiple, {
      props
    })

    const autocomplete = wrapper.findComponent(AutoComplete)

    // Simulate a search event
    await autocomplete.vm.$emit('complete', { query: 'A' })

    expect(wrapper.vm.items).toEqual([{ name: 'Apple' }])
  })
})
