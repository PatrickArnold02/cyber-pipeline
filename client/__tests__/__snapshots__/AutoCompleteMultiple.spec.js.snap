// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AutoCompleteMultiple > matches the snapshot 1`] = `
"<div data-v-8f5fbc77="" class="w-full">
  <!-- Floating Label--><span data-v-8f5fbc77="" class="p-floatlabel p-floatlabel-over w-full" data-pc-name="floatlabel" pc19="" data-pc-section="root"><!-- Icon --><div data-v-8f5fbc77="" class="p-iconfield w-full" iconposition="left" data-pc-name="iconfield" pc20="" data-pc-section="root"><span data-v-8f5fbc77="" class="p-inputicon" data-pc-name="inputicon" pc21="" data-pc-section="root"><i data-v-8f5fbc77="" class="pi pi-search"></i></span><!-- Autocomplete Component-->
  <div data-v-8f5fbc77="" class="p-autocomplete p-component p-inputwrapper p-invalid w-full" style="position: relative;" id="testField" data-pc-name="autocomplete" pc22="" data-pc-section="root">
    <!---->
    <ul class="p-autocomplete-input-multiple" tabindex="-1" role="listbox" aria-orientation="horizontal" data-pc-section="inputmultiple">
      <li class="p-autocomplete-input-chip" role="option" data-pc-section="inputchip"><input type="text" class="" tabindex="0" autocomplete="off" role="combobox" aria-haspopup="listbox" aria-autocomplete="list" aria-expanded="false" aria-controls="testField_list" aria-invalid="true" data-pc-section="input"></li>
    </ul>
    <!---->
    <!----><span role="status" aria-live="polite" class="p-hidden-accessible" data-pc-section="hiddensearchresult" data-p-hidden-accessible="true">No results found</span>
    <!--teleport start-->
    <!--teleport end-->
  </div>
</div><!-- Label --><label data-v-8f5fbc77="" for="testField" class="ml-20">Test Label</label></span><!-- Error Text--><small data-v-8f5fbc77="" id="testField-help" class="w-full text-red-600">Field is required</small></div>"
`;
